src/coreclr/tests/src/tracing/eventpipe/reverse/reverse.cs

```diff
@@ -0,0 +1,273 @@
+        }
+
+        public static async Task<bool> TEST_ReverseConnectionCanRecycleWhileTracing()
+        {
+            string serverName = ReverseServer.MakeServerAddress();
+            Logger.logger.Log($"Server name is '{serverName}'");
+            await RunSubprocess(
+                serverName: serverName,
+                duringExecution: async (int pid) =>
+                {
+                    Task regularTask = Task.Run(async () => 
+                    {
+                        var config = new SessionConfiguration(
+                            circularBufferSizeMB: 1000,
+                            format: EventPipeSerializationFormat.NetTrace,
+                            providers: new List<Provider> { 
+                                new Provider("Microsoft-DotNETCore-SampleProfiler")
+                            });
+                        Logger.logger.Log("Starting EventPipeSession over standard connection");
+                        using Stream stream = EventPipeClient.CollectTracing(pid, config, out var sessionId);
+                        Logger.logger.Log($"Started EventPipeSession over standard connection with session id: 0x{sessionId:x}");
+                        using var source = new EventPipeEventSource(stream);
+                        Task readerTask = Task.Run(() => source.Process());
+                        await Task.Delay(500);
+                        Logger.logger.Log("Stopping EventPipeSession over standard connection");
+                        EventPipeClient.StopTracing(pid, sessionId);
+                        await readerTask;
+                        Logger.logger.Log("Stopped EventPipeSession over standard connection");
+                    });
+
+                    Task reverseTask = Task.Run(async () => 
+                    {
+                        var ad1 = await WaitTillTimeout(ReverseServer.CreateServerAndReceiveAdvertisement(serverName), TimeSpan.FromMilliseconds(_maxPollTimeMS));
+                        Logger.logger.Log(ad1.ToString());
+                        var ad2 = await WaitTillTimeout(ReverseServer.CreateServerAndReceiveAdvertisement(serverName), TimeSpan.FromMilliseconds(_maxPollTimeMS));
+                        Logger.logger.Log(ad2.ToString());
+                        var ad3 = await WaitTillTimeout(ReverseServer.CreateServerAndReceiveAdvertisement(serverName), TimeSpan.FromMilliseconds(_maxPollTimeMS));
+                        Logger.logger.Log(ad3.ToString());
+                        var ad4 = await WaitTillTimeout(ReverseServer.CreateServerAndReceiveAdvertisement(serverName), TimeSpan.FromMilliseconds(_maxPollTimeMS));
+                        Logger.logger.Log(ad4.ToString());
+                    });
+
+                    await Task.WhenAll(reverseTask, regularTask);
+                }
+            );
+
+            return true;
+        }
+
+        public static async Task<bool> TEST_StandardConnectionStillWorksIfReverseConnectionIsBroken()
```

Comment:
Scenario (b) should already be tested since the mock reverse server doesn't send any data to the target process in these tests.  It just accepts, reads the advertise, and then either stays open or closes.

Reply:
What test case does it stay open in? Every case I noticed if the server accepted the request then it either responded or closed the socket.

Reply:
The `ReverseServer` never sends any commands to the target.  It creates the socket, accepts, and then either closes immediately or stays open.  I have a test where the socket is created before the target process and stays open, but is closed while the target process is alive.  I _should_ add a test where the socket remains open till after the target process closes, though.